odoo/osv/expression.py

```diff
@@ -337,6 +337,20 @@ def _quote(to_quote):
     return to_quote
 
 
+def _shorten_alias(alias):
+    # Use an alternate alias scheme if length exceeds the PostgreSQL limit
+    # of 63 characters.
+    if len(alias) >= 64:
+    # We have to fit a crc32 hash and one underscore
+    # into a 63 character alias. The remaining space we can use to add
+    # a human readable prefix.
+        alias_hash = hex(crc32(alias.encode('utf-8')))[2:]
+        ALIAS_PREFIX_LENGTH = 63 - len(alias_hash) - 1
+        alias = "%s_%s" % (
```

Comment:
Since this is being moved / rewritten, we could use `%08x` for the formatting of the `alias_hash` (which would be just the result of the crc32 call without `hex()`), and slice the alias at 54 (63 - 8 for hex'd CRC32 - 1 for the underscore).

Reply:
This should work, we've tested this successfully

```python
def _shorten_alias(alias):
    # Use an alternate alias scheme if length exceeds the PostgreSQL limit
    # of 63 characters.
    if len(alias) >= 64:
        # We have to fit a crc32 hash and one underscore
        # into a 63 character alias. The remaining space we can use to add
        # a human readable prefix.
        alias = "%s_%08x" % (alias[:54], crc32(alias.encode('utf-8')))
    return alias
```

Note I've also removed the [:2] for the crc32, I assume that was used to remove part of the signed crc values in Python 2. Per the current [Zlib Docs](https://docs.python.org/3/library/zlib.html) this is no longer the case in Python 3

> Changed in version 3.0: Always returns an unsigned value. To generate the same numeric value across all Python versions and platforms, use crc32(data) & 0xffffffff.

Reply:
> Note I've also removed the [:2] for the crc32

Good.

> I assume that was used to remove part of the signed crc values in Python 2

It was used to remove the `0x` prefix, `hex()` returns `"0x<...>"` while `%x` just returns the nibbles with no prefix:
```python
>>> hex(42)
'0x2a'
>>> format(42, 'x')
'2a'
```
It probably just behaved oddly with negative crc (it would slice off the `-0` and leave the `x` part of the prefix, which wasn't stringly harmful but would "waste" one byte).

Reply:
@alexhafner Odoo 13.0 requires Python 3.6 or above.
So you can go for the code using the string format `"%s_%08x"`.